tests/page-objects/default/common/common.wdio.page.js

```diff
@@ -146,9 +140,22 @@ const hideSnackbar = () => {
 
 const getVisibleLoaders = async () => {
   const visible = [];
-  for (const loader of await $$('.container-fluid .loader')) {
-    if (await loader.isDisplayedInViewport()) {
-      visible.push(loader);
+
+  // Get all loaders in the page
+  const loaders = await $$('.loader').getElements();
+  if (loaders.length) {
+    // Add a small pause to let the DOM stabilize
+    await browser.pause(100);
+    // Instead of iterating through the loaders array, query for each loader individually
+    // This avoids issues with stale references
+    for (let i = 0; i < loaders.length; i++) {
+      // Use a more specific selector to get a fresh reference to each loader
+      // This avoids the index out of bounds issue
+      const loaderSelector = `.loader:nth-child(${i + 1})`;
```

Comment:
I don't think this will work. 
`nth-child` matches elements based on the indexes of the elements in the child list of their parents (reference: https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child)
So this won't be matching your 4th existent loader, it will be matching _a loader_ that is the 4th element in its parent.

Reply:
I changed it to `nth-of-type`, what do you think about it? What I understand about this selector is for example: `.loader:nth-of-type(1) ` would select the first loader because it's the 1st element with class "loader"

Reply:
Thank you so much for catching this.

Reply:
Sweet! I think `nth-of-type` is the way to go here!